Stabilize message, new option, switch to defineMessages

ericf opened the pull request:
This stabilizes the `defaultMessage` by parsing and pretty-printing it; this removes any insignificant whitespace, and should yield less false-positive re-translations.

This adds a new `enforceDescriptions` option to require that all declared messages _must_ contain a `description` for translators.

The `description` field will now _always_ be removed from the source code to save k-weight.

This also switches from `defineMessage` to the new `defineMessages`.

ericf commented:
I'm open to bikeshedding these option names.

mridgway commented:
lgtm

caridy commented:
+1

_note: since the method is called `defineMessages()` maybe we can use `generateIds` instead of `generateMessageIds`, it seems redundant, jejejeje, a little bit of bikeshedding here... sorry._

ericf commented:
@mridgway @caridy I've changed the scope of the PR and removed the message id generation (at least for now). I feel that the id generation is likely a footgun, so I want to continue exploring that space, but not let it block progress. See the updated PR description for the new details.

mridgway commented:
@ericf Sounds good. ID generation does indeed have a lot of edge cases and is not high on our team's priority. :+1:

caridy commented:
ok
